### ISSUE TITLE
listStaticSiteSecrets results are not marked as secrets

### ISSUE THREAD

**Issue by TechWatching** [pulumi/pulumi-azure-native#2408]
### What happened?

The [azure-native.web.listStaticSiteSecrets](https://www.pulumi.com/registry/packages/azure-native/api-docs/web/liststaticsitesecrets/) methods return a result that is not maked as a secret so it appear in plain text when outputing it. 

### Expected Behavior

The listStaticSiteSecrets result should be marked as a secret. 

### Steps to reproduce

```ts
const staticWebApp = new web.StaticSite("stapp-vue2048-preprod", {
    resourceGroupName: resourceGroup.name,
    sku: {
        name: "Free",
        tier: "Free"
    },
    repositoryUrl: "",
});

export const staticWebAppDeployToken = pulumi.secret(web.listStaticSiteSecretsOutput({name: staticWebApp.name, resourceGroupName: resourceGroup.name}).properties["apiKey"]);
```
If we remove the pulumi.secret the output is not a secret

### Output of `pulumi about`

CLI
Version      3.64.0
Go Version   go1.20.3
Go Compiler  gc

Plugins
NAME          VERSION
azure-native  1.100.1
nodejs        unknown

Host
OS       Microsoft Windows 11 Enterprise
Version  10.0.22000 Build 22000
Arch     x86_64

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by rquitales** [comment]
@TechWatching Thanks for reporting this issue. Yes, these responses should be marked as a secret within our provider so they can be generated as a secret field!

**Comment by TechWatching** [comment]
@mnlumi @rquitales Is it something coming soon? Currently anyone that does use the azure-typescript template (or similar template creating a storage account and listing its keys) will output in clear text the secret. If someone tries to run this template in a GitHub Actions workflow, the secret will be available in the logs.

By the way, the `listStaticSiteSecrets`function is not the only one with this issue.

**Comment by mnlumi** [comment]
hey @TechWatching. I will work with the team to prioritize this work and update you as soon I have news. Thanks for checking in!

**Comment by t0yv0** [comment]
Related https://github.com/pulumi/pulumi/issues/12710

**Comment by TechWatching** [comment]
@t0yv0 @mikhailshilkov  As the related issue has been completed, could this be something considered?

**Comment by t0yv0** [comment]
Tracked in https://github.com/pulumi/pulumi-terraform-bridge/issues/1051 unfortunately there are still some backwards-compatibility obstacles we need to plan for.

**Comment by mikhailshilkov** [comment]
@t0yv0 Are you sure the bridge issue is right to track work in Azure Native?

**Comment by t0yv0** [comment]
Good point Mikhail, thanks for pointing this out! We might want to move it to pulumi/pulumi, but I believe the safety considerations still apply so it is part of the same workstream. As things are written, editing azure-native to emit secret markers will break existing programs, we wanted to downgrade the hard break to something like a comptime warning for the rollout first.
